internal/pkg/kube/reconciler_manager.go

```diff
@@ -42,12 +43,13 @@ func NewReconcilerManager(pushRequests <-chan xds.PushRequest, reconcilers ...Re
 }
 
 func (rm *ReconcilerManager) ReconcileAll(ctx context.Context) error {
+	var reconcileErrs []error
```

Comment:
Nit: We could avoid potential reallocation by specifying a capacity.

```suggestion
	reconcileErrs := make([]error, 0, len(rm.reconcilers))
```

Reply:
We should probably make it `make([]error, 0, len(rm.reconcilers)))`, so the length is set correctly to `0` but capacity is reflecting how many calls we are going to make. Otherwise `append` will immediately result in doubling the capacity and so creating the new underlying array IIRC.

However, more often than not I see an nil slice used for such simple cases. I'm fine either way though. WDYT @jewertow?

Reply:
> WDYT @jewertow?

I am open to these kinds of micro optimizations. I did not pay attention to such details, because it was irrelevant in a poc, but if we can do something better, why not?